src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java

```diff
@@ -69,6 +69,63 @@ public static DocstringInfo parseDocstring(String doc, List<DocstringParseError>
     return result;
   }
 
+  /**
+   * Removes leading whitespace from docstring lines after the first.
+   *
+   * Allows docstrings to be indented to the context where they appear, without including that indent in the output.
+   * Some output formats have semantic meaning for leading whitespace (such as Markdown).
+   *
+   * @param doc A non-function docstring, without Args or Returns sections.
+   * 
+   */
+  public static String dedentDocstring(String doc) {
+    // Two iterations: first buffer lines and find the minimal indent, then trim to the min
+    List<String> buffer = new ArrayList<>();
+    int indentation = Integer.MAX_VALUE;
+    int lineOffset = 0;
+    int endOfLineOffset = 0;
+
+    do {
+      endOfLineOffset = doc.indexOf("\n", lineOffset);
+      String line = endOfLineOffset < 0 ? doc.substring(lineOffset) : doc.substring(lineOffset, endOfLineOffset + 1);
+      boolean allWhitespace = line.trim().isEmpty();
```

Comment:
`line.trim()` removes tabs (as well as all other characters below 0x20, e.g. windows carriage returns), not just spaces. `DocstringParser.getIndentation()` counts only spaces.